charts/jaeger/templates/collector-deploy.yaml

```diff
@@ -32,6 +32,20 @@ spec:
         {{- toYaml .Values.collector.podLabels | nindent 8 }}
 {{- end }}
     spec:
+      {{- if .Values.provisionDataStore.elasticsearch }}
+      initContainers:
+        - name: elasticsearch-checker
+          image: curlimages/curl
+          command: 
+          - sh
+          - "-c"
+          - |
+            url="{{ include "elasticsearch.client.url" . }}"
+            until curl -s "$url/_cluster/health?wait_for_status=green"; do
```

Comment:
`until` checks process exit code. Should you not be checking HTTP status code instead?